packages/odf/modals/s3-browser/delete-and-empty-bucket/DeleteBucketModal.tsx

```diff
@@ -0,0 +1,210 @@
+import * as React from 'react';
+import {
+  BUCKET_BOOKMARKS_USER_SETTINGS_KEY,
+  LIST_VERSIONED_OBJECTS,
+} from '@odf/core/constants';
+import { useUserSettingsLocalStorage } from '@odf/shared';
+import { CommonModalProps } from '@odf/shared/modals';
+import { S3Commands } from '@odf/shared/s3';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { LaunchModal } from '@openshift-console/dynamic-plugin-sdk/lib/app/modal-support/ModalProvider';
+import { Trans } from 'react-i18next';
+import useSWR from 'swr';
+import {
+  Modal,
+  Button,
+  Text,
+  TextInput,
+  ModalVariant,
+  ValidatedOptions,
+  TextInputTypes,
+  TextVariants,
+  TextContent,
+  FormGroup,
+  Alert,
+  Spinner,
+  AlertVariant,
+  AlertActionLink,
+  ButtonVariant,
+} from '@patternfly/react-core';
+import { EmptyBucketResponse, getTextInputLabel } from './EmptyBucketModal';
+import { LazyEmptyBucketModal } from './lazy-delete-and-empty-bucket';
+
+type DeleteBucketModalProps = {
+  bucketName: string;
+  noobaaS3: S3Commands;
+  launcher: LaunchModal;
+  refreshTokens?: () => void;
+  setEmptyBucketResponse: React.Dispatch<
+    React.SetStateAction<EmptyBucketResponse>
+  >;
+};
+
+const DeleteBucketModal: React.FC<CommonModalProps<DeleteBucketModalProps>> = ({
+  closeModal,
+  isOpen,
+  extraProps: {
+    bucketName,
+    noobaaS3,
+    launcher,
+    refreshTokens,
+    setEmptyBucketResponse,
+  },
+}) => {
+  const { t } = useCustomTranslation();
+  const [inputValue, setInputValue] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const {
+    data,
+    error,
+    isLoading: isChecking,
+  } = useSWR(`${bucketName}-${LIST_VERSIONED_OBJECTS}`, () =>
+    noobaaS3.listObjectVersions({ Bucket: bucketName, MaxKeys: 1 })
+  );
+  const hasObjects =
+    data?.Versions?.length > 0 || data?.DeleteMarkers?.length > 0;
+
+  const [_favorites, setFavorites] = useUserSettingsLocalStorage<string[]>(
+    BUCKET_BOOKMARKS_USER_SETTINGS_KEY,
+    true,
+    []
+  );
+
+  const onDelete = async (event) => {
+    event.preventDefault();
+    setInProgress(true);
+
+    try {
+      await noobaaS3.deleteBucket({
+        Bucket: bucketName,
+      });
+
+      setInProgress(false);
+      closeModal();
+      setFavorites((oldFavorites) =>
+        oldFavorites.filter((bucket) => bucket !== bucketName)
+      );
+      refreshTokens?.();
+    } catch {
+      setInProgress(false);
+    }
+  };
```

Comment:
what if `deleteBucket` fails ?? u need to store error for that case...
```suggestion
    try {
      await noobaaS3.deleteBucket({
        Bucket: bucketName,
      });

      setInProgress(false);
      closeModal();
      setFavorites((oldFavorites) =>
        oldFavorites.filter((bucket) => bucket !== bucketName)
      );
      refreshTokens?.();
    } catch(err) {
      setInProgress(false);
      setDeleteError(err);
    }
  };
```